--- main.go	2023-10-05 14:07:54.808446767 -0700
+++ mainpatch.go	2023-10-05 14:08:28.426592750 -0700
@@ -34,6 +34,8 @@
 	}

-	cmd := otelcol.NewCommand(params)
+	cmd := otelcol.NewCommandMustSetProvider(params)
+	// this is only here so that the flag can be recognized by the upstream parser and displayed in help
+	addOpampConfigFlag(cmd.Flags())
 	if err := cmd.Execute(); err != nil {
 		log.Fatalf("collector server run finished with error: %v", err)
 	}
