--- main_windows.go.orig	2024-05-21 20:10:47.341714100 +0000
+++ main_windows.go	2024-05-21 20:10:59.675047700 +0000
@@ -15,6 +15,11 @@
 func run(params otelcol.CollectorSettings) error {
 	// No need to supply service name when startup is invoked through
 	// the Service Control Manager directly.
+	err := UseCustomConfigProvider(&params)
+	if err != nil {
+		return err
+	}
+
 	if err := svc.Run("", otelcol.NewSvcHandler(params)); err != nil {
 		if errors.Is(err, windows.ERROR_FAILED_SERVICE_CONTROLLER_CONNECT) {
 			// Per https://learn.microsoft.com/en-us/windows/win32/api/winsvc/nf-winsvc-startservicectrldispatchera#return-value
